prov/rxm/src/rxm_ep.c

```diff
@@ -512,13 +516,21 @@ static ssize_t rxm_ep_send_common(struct fid_ep *ep_fid, const struct iovec *iov
 		pkt_size = sizeof(*pkt) + pkt->hdr.size;
 	}
 
-	ret = fi_send(rxm_conn->msg_ep, pkt, pkt_size, desc_tx_buf, 0, tx_entry);
-	if (ret) {
-		FI_WARN(&rxm_prov, FI_LOG_EP_DATA, "fi_send for MSG provider failed\n");
-		goto err;
+	if (flags & FI_INJECT & ~FI_COMPLETION) {
+		ret = fi_inject(rxm_conn->msg_ep, pkt, pkt_size, 0);
```

Comment:
I have investigated some approaches how to handle case when user expects no CQ on the operation.
I read man pages related to FI_SELECTIVE_COMPLETION and man pages say that it is user responsibility to take care of _when a request that does NOT have FI_COMPLETION set has completed indirectly_.
In new patch set I drop the flag FI_COMPLETION from this check, because it was incorrectly to release pre-allocated buffer from pool before operation has completed (since new transmission operation will pop this buffer). But in current implementation, if user won't invoke fi_cq_read at all, the pool of pre-allocated transmission buffer will be exhausted and user won't be able to send new data. And moreover, since rxm doesn't pass flags to send (underlying provider uses its own op_flags), it may cause that generated CQs by underlying provider are never read by rxm.

As I mentioned above, I'm going just drop FI_COMPLETION flag from if statement.
Regarding max inject_size of transmission context, I checked socket provider (MSG type of endpoint) and I guess we can avoid extra check in rxm, because socket provider compares its own max_inject_size and size of passed data. Do you agree with me that RXM without this check is OK?